packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts

```diff
@@ -0,0 +1,232 @@
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { ethers } from 'ethers'
+import { hex } from '@scure/base'
+import { p2tr, p2tr_ns, taprootTweakPubkey, TAPROOT_UNSPENDABLE_KEY } from '@scure/btc-signer'
+import { BaseEndpointTypes } from '../endpoint/proof-of-reserves'
+import abi from '../config/dlc-manager-abi.json'
+import {
+  BitcoinNetwork,
+  BitcoinTransaction,
+  BitcoinTransactionVectorOutput,
+  getBitcoinNetwork,
+  RawVault,
+} from './utils'
+
+const logger = makeLogger('dlcBTC PoR')
+
+export type TransportTypes = BaseEndpointTypes
+
+export class DLCBTCPorTransport extends SubscriptionTransport<TransportTypes> {
+  name!: string
+  requester!: Requester
+  provider!: ethers.providers.JsonRpcProvider
+  dlcManagerContract!: ethers.Contract
+  settings!: TransportTypes['Settings']
+
+  async initialize(
+    dependencies: TransportDependencies<TransportTypes>,
+    adapterSettings: TransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    const { RPC_URL, CHAIN_ID, DLC_CONTRACT } = adapterSettings
+    this.settings = adapterSettings
+    this.provider = new ethers.providers.JsonRpcProvider(RPC_URL, CHAIN_ID)
+    this.dlcManagerContract = new ethers.Contract(DLC_CONTRACT, abi, this.provider)
+    this.requester = dependencies.requester
+  }
+  async backgroundHandler(context: EndpointContext<TransportTypes>) {
+    await this.handleRequest()
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest() {
+    let response: AdapterResponse<TransportTypes['Response']>
+    try {
+      response = await this._handleRequest()
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: [], response }])
+  }
+
+  async _handleRequest(): Promise<AdapterResponse<TransportTypes['Response']>> {
+    const providerDataRequestedUnixMs = Date.now()
+
+    // Get all vault data. Filter placeholder values.
+    const vaultData: RawVault[] = (
+      await this.dlcManagerContract.getFundedDLCs(0, this.settings.MAX_VAULTS)
+    ).filter(
+      (v: RawVault) =>
+        v.uuid != '0x0000000000000000000000000000000000000000000000000000000000000000',
+    )
+
+    // Get the Attestor Public Key
+    const attestorPublicKey = await this.dlcManagerContract.attestorGroupPubKey()
+
+    let totalPoR = 0
+    const concurencyGroupSize =
```

Comment:
[low priority] can this just be 
`const concurencyGroupSize = this.settings.GROUP_SIZE || vaultData.length`?